backend/src/CardHandler.ts

```diff
@@ -0,0 +1,373 @@
+import { GameCard, CardType } from './types/Cards';
+import { SourceKind, PickCardEvent, PlayerUpdateEvent } from './types/Events';
+import { Player } from './types/Player';
+import { Game } from './types/Game';
+
+export const handlePlayCard = (card: GameCard, source: Player, target: Player, game: Game) => {
+  switch (card.type) {
+    case CardType.BANG:
+      return [new PickCardEvent({
```

Comment:
I think these should be events that fire after the card is certain to go through
so prompting the player to give a missed would come earlier (because we'll have to handle barrels as well)

Reply:
so here would just be a PlayerUpdateEvent to reduce health